internal/commands/user/revoke.go

```diff
@@ -0,0 +1,151 @@
+package user
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/10gen/realm-cli/internal/app"
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/10gen/realm-cli/internal/utils/flags"
+	"github.com/spf13/pflag"
+)
+
+// CommandRevoke is the `user revoke` command
+type CommandRevoke struct {
+	inputs      revokeInputs
+	outputs     []userOutput
+	realmClient realm.Client
+}
+
+type revokeInputs struct {
+	app.ProjectInputs
+	usersInputs
+}
+
+func (i *revokeInputs) Resolve(profile *cli.Profile, ui terminal.UI) error {
+	if err := i.ProjectInputs.Resolve(ui, profile.WorkingDirectory); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Flags is the command flags
+func (cmd *CommandRevoke) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+	fs.StringSliceVarP(&cmd.inputs.Users, flagUser, flagUserShort, []string{}, flagUserRevokeUsage)
+	fs.VarP(
+		flags.NewEnumSet(&cmd.inputs.ProviderTypes, validAuthProviderTypes()),
+		flagProvider,
+		flagProviderShort,
+		flagProviderUsage,
+	)
+	fs.VarP(&cmd.inputs.State, flagState, flagStateShort, flagStateUsage)
+	fs.BoolVarP(&cmd.inputs.Pending, flagPending, flagPendingShort, false, flagPendingUsage)
+}
+
+// Inputs is the command inputs
+func (cmd *CommandRevoke) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Setup is the command setup
+func (cmd *CommandRevoke) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = profile.RealmAuthClient()
+	return nil
+}
+
+// Handler is the command handler
+func (cmd *CommandRevoke) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, appErr := app.Resolve(ui, cmd.realmClient, cmd.inputs.Filter())
+	if appErr != nil {
+		return appErr
+	}
+	users, usersErr := cmd.inputs.ResolveUsers(ui, cmd.realmClient, app)
+	if usersErr != nil {
+		return usersErr
+	}
+	for _, user := range users {
+		err := cmd.realmClient.RevokeUserSessions(app.GroupID, app.ID, user.ID)
+		cmd.outputs = append(cmd.outputs, userOutput{user: user, err: err})
+	}
+	return nil
+}
+
+// Feedback is the command feedback
+func (cmd *CommandRevoke) Feedback(profile *cli.Profile, ui terminal.UI) error {
+	if len(cmd.outputs) == 0 {
+		return ui.Print(terminal.NewTextLog("No users to revoke, try changing the --user input"))
```

Comment:
I came across an interesting scenario here from my own user error (though I could see another dev doing the same). When selecting the users with a multi-select, I tend to press enter instead of space and accidentally select no users. I modified this message here as sort of an indicator that maybe the user did not select a user. We use text like `No users to revoke` elsewhere but I wanted to get some thoughts on the messaging here since the shorter text could not be registered as a user did not select any users and just pressed enter.

Reply:
I like the change, I think we can catch 0 users selected when in interactive mode and print `No users were selected to revoke` and end the command there

Reply:
That's a good idea. I will take a look around and think about this some more. The one tricky part is that there is a difference between no users found and no users selected and only the common inputs.go code would be able to tell the difference. I know we previously discussed having each command have it's own logic on what to do with the found users (which I still think is a good approach). But maybe the `ResolveUsers` function should be split between finding users and selecting users? That would let the individual commands dictate the flow

Reply:
I decided to stick with the message in feedback since I wasn't sure if we wanted to display an error or just a log like this if a user selects no users. Wanted to get some more thoughts here, but if we stick with what is in the PR, I can add the `try changing the -user input` text to the other commands

Reply:
My preference is leaning towards the display an error, but I also don't have a super strong opinion if @makesitgo feels otherwise.

Reply:
i think ideally we aren't ever printing any `ERROR` messages in command code (and let those be entirely system driven).  in this case, i think just printing `No users to revoke sessions for` (or `No users found...`) is just fine